pinot-spi/src/main/java/org/apache/pinot/spi/stream/OffsetCriteria.java

```diff
@@ -172,17 +196,21 @@ public OffsetCriteria withOffsetString(String offsetString) {
         _offsetCriteria.setOffsetType(OffsetType.SMALLEST);
       } else if (offsetString.equalsIgnoreCase(OffsetType.LARGEST.toString())) {
         _offsetCriteria.setOffsetType(OffsetType.LARGEST);
-      } else {
+      } else if (TimeUtils.isPeriodValid(offsetString)) {
         try {
-          Long periodToMillis = TimeUtils.convertPeriodToMillis(offsetString);
+          long periodToMillis = TimeUtils.convertPeriodToMillis(offsetString);
           if (periodToMillis >= 0) {
             _offsetCriteria.setOffsetType(OffsetType.PERIOD);
           } else {
             _offsetCriteria.setOffsetType(OffsetType.CUSTOM);
           }
-        } catch (Exception e) {
-          _offsetCriteria.setOffsetType(OffsetType.CUSTOM);
+        } catch (Exception ignore) {
```

Comment:
Since we already check if the period is valid, we should just remove the try-catch